test/rpc-test.js

```diff
@@ -0,0 +1,227 @@
+/* eslint-env mocha */
+/* eslint prefer-arrow-callback: "off" */
+
+'use strict';
+
+const assert = require('./util/assert');
+const consensus = require('../lib/protocol/consensus');
+const FullNode = require('../lib/node/fullnode');
+const Network = require('../lib/protocol/network');
+const network = Network.get('regtest');
+
+const node = new FullNode({
+  network: 'regtest',
+  apiKey: 'foo',
+  walletAuth: true,
+  memory: true,
+  workers: true,
+  plugins: [require('../lib/wallet/plugin')]
+});
+
+const {NodeClient, WalletClient} = require('bclient');
+
+const nclient = new NodeClient({
+  port: network.rpcPort,
+  apiKey: 'foo'
+});
+
+const wclient = new WalletClient({
+  port: network.walletPort,
+  apiKey: 'foo'
+});
+
+const {wdb} = node.require('walletdb');
+
+let addressHot = null;
+let addressMiner = null;
+let walletHot = null;
+let walletMiner = null;
+let blocks = null;
+let txid = null;
+let utxo = null;
+
+describe('RPC', function() {
+  this.timeout(15000);
+
+  it('should open node and create wallets', async () => {
+    consensus.COINBASE_MATURITY = 0;
+    await node.open();
+    await nclient.open();
+    await wclient.open();
+
+    const walletHotInfo = await wclient.createWallet('hot');
+    walletHot = wclient.wallet('hot', walletHotInfo.token);
+    const walletMinerInfo = await wclient.createWallet('miner');
+    walletMiner = wclient.wallet('miner', walletMinerInfo.token);
+    await walletHot.open();
+    await walletMiner.open();
+  });
+
+  it('should rpc help', async () => {
+    assert( await nclient.execute('help', []) );
+    assert( await wclient.execute('help', []) );
+
+    await assert.asyncThrows( async () => await nclient.execute('help',
+      ['getinfo']), 'getinfo');
+
+    await assert.asyncThrows( async () => await wclient.execute('help',
+      ['getbalance']), 'getbalance');
+  });
+
+  it('should rpc getinfo', async () => {
+    const info = await nclient.execute('getinfo', []);
+    assert.strictEqual(info.blocks, 0);
+  });
+
+  it('should rpc selectwallet', async () => {
+    const response = await wclient.execute('selectwallet', ['miner']);
+    assert.strictEqual(response, null);
+  });
+
+  it('should rpc getnewaddress from default account', async () => {
+    const acctAddr = await wclient.execute('getnewaddress', []);
+    assert(acctAddr);
```

Comment:
Any more assertions, for example it's for the correct network?

Reply:
Actually having some issues with the notorious `Address` module network issues (everything is main, even when its not...) I think I'm going to pick at that module and maybe make a PR for better network properties of `Address`, then can come back to this and write a good test...

Reply:
Correct network can be easily tested after merge of https://github.com/bcoin-org/bcoin/pull/720

Reply:
Ok for now gunna test with `assert(Address.fromString(acctAddr.toString()));` which should at least catch a regression if the RPC call returns junk